php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -169,6 +169,8 @@ class NDB_BVL_Instrument extends NDB_Page
 
         // Ensure 'page' is a string.
         $page = $page ?? '';
+        // Ensure 'commentID' is a string.
+        $commentID = $commentID ?? '';
```

Comment:
@ridz1208 I think this would be better written in the function signature, i.e.

```php
static function factory($instrument, $commentID = '', $page = '', $guarantee_exists = true)
```

I think I wrote the original line on 171 and realized it was silly but have been too busy to fix it. 😅 Let's address it now.

Reply:
@johnsaigle done

Reply:
@johnsaigle I reverted back to this change for the following reason. If in the function call, the `null` argument is explicitely passed to the parameter `$commentID` and/or `$page` then even if you declare it as `$commentID = '', $page = ''`  php still sets it as `null` thus the NULL coalesce is required until we go over all the code calling this function and omit an expicit null or convert them to `''`